### PR TITLE
Using telemetry to provide traceability to nodes

### PR DESCRIPTION
**Pull request by netsirius**
My purpose is to give traceability on everything that happens within a node, especially on the operations.

The main idea is to create root span by initializing the node and trace everything that happens inside the node as children spans inside the root span.

image

I have been testing this from the operations tests, but I have found the problem that the spans created in a different threads than where the tracer has been defined are not created within the root span.

### CODE REVIEW

File: crates/locutus-node/src/node/tracer.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+use opentelemetry::sdk::propagation::TraceContextPropagator;
+use tracing_opentelemetry::OpenTelemetryLayer;
+use tracing_subscriber::layer::Layered;
+use tracing_subscriber::prelude::*;
+use tracing_subscriber::Registry;
+
+pub fn init_tracer(service_name: String) -> Result<(), TraceError> {
+    let tracer = opentelemetry_jaeger::new_pipeline()
+        .with_service_name(service_name)
+        //.with_agent_endpoint("127.0.0.1:6831")
```
Comment (iduartgomez) [inline crates/locutus-node/src/node/tracer.rs:12]: remember to pass this via config (we can add a new environment variable for now)

**Comment by iduartgomez** [comment]
@netsirius we can merge this as it is (or with whatever else changes you need to do) and will fix remaining issues later on if you prefer to do it that way.

**Comment by netsirius** [comment]
@iduartgomez I have placed the initialize method next to the logger and set the dependencies as optional features.

I have also tried to test what we are talking about starting the subscriber before launching the different asynchronous tasks, but I haven't obtained the expected result.

The tracer service name, the Jaeger agent hostname and many properties can be defined using environment variables.

**Review by iduartgomez — APPROVED** [review]

**Comment by iduartgomez** [comment]
There are some compile errors with the PR @netsirius

**Comment by netsirius** [comment]
@iduartgomez I have added that everything related to tracing is only used when the trace feature is being used

**Review by iduartgomez — APPROVED** [review]

**Pull request merged**